source/plugins/go/input/lib/proxyutils.go

```diff
@@ -0,0 +1,91 @@
+package lib
+
+import (
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
+var proxyCertPath = "/etc/ama-logs-secret/PROXYCERT.crt"
+
+type ProxyConfiguration struct {
+	User string
+	Pass string
+	Addr string
+	Port string
+}
+
+// func getProxyConfiguration() map[string]string {
+// 	amalogsproxySecretPath := "/etc/ama-logs-secret/PROXY"
+// 	if _, err := os.Stat(amalogsproxySecretPath); os.IsNotExist(err) {
+// 		return make(map[string]string)
+// 	}
+
+// 	proxyConfig, err := parseProxyConfiguration(amalogsproxySecretPath)
+// 	if err != nil {
+// 		// todo: handle error
+// 		return make(map[string]string)
+// 	}
+
+// 	// todo: log if empty
+// 	return proxyConfig
+// }
+
+func getProxyEndpoint() string {
+	amaLogsProxySecretPath := "/etc/ama-logs-secret/PROXY"
+	proxyConfig, err := ioutil.ReadFile(amaLogsProxySecretPath)
+	if err != nil {
+		// todo: handle error
+		return ""
+	}
+	return strings.TrimSpace(string(proxyConfig))
+}
+
+// func parseProxyConfiguration(proxyConfPath string) (map[string]string, error) {
```

Comment:
Looks like this code needs to be uncommented. why this code being commented?

Reply:
in ruby we need to parse components of proxy url for the call 
`http_client = Net::HTTP.new(@post_request_uri.host, @post_request_uri.port, @proxy[:addr], @proxy[:port], @proxy[:user], @proxy[:pass])
`
but in golang we dont need to do so

```
			proxyEndpointUrl, err := url.Parse(proxyEndpoint)
			if err != nil {
				aiLogger.Printf("Error parsing proxy endpoint: %s", err.Error())
				telemetryClient = nil
				return
			}

			transport := &http.Transport{
				Proxy: http.ProxyURL(proxyEndpointUrl),
			}
			httpClient := &http.Client{
				Transport: transport,
			}
```